server/utils/errors.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"errors"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	argoerrors "github.com/argoproj/argo-workflows/v3/errors"
+
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// Return a new status error and true else nil and false
+// is meant to be as close to the opposite of grpc gateways status -> http error code converter
+// exceptions made where one to one mappings were not available
+func httpToStatusError(code int, msg string) (error, bool) {
+	// handle success & information  codes in one go
+	if code < 300 {
+		return status.Error(codes.OK, msg), true
+	}
+
+	errorMaps := map[int]codes.Code{
```

Comment:
@isubasinghe Can you move to constants instead of initializing every call?

Reply:
done